llama-index-integrations/vector_stores/llama-index-vector-stores-redis/llama_index/vector_stores/redis/base.py

```diff
@@ -18,122 +18,110 @@
+    _doc_id_field_name: str = "doc_id"
+    _text_field_name: str = "text"
 
     def __init__(
         self,
-        index_name: str,
-        index_prefix: str = "llama_index",
-        prefix_ending: str = "/vector",
-        index_args: Optional[Dict[str, Any]] = None,
-        metadata_fields: Optional[List[str]] = None,
-        redis_url: str = "redis://localhost:6379",
+        schema: Optional[IndexSchema] = None,
+        redis_client: Optional[Redis] = None,
+        redis_url: Optional[str] = None,
         overwrite: bool = False,
         **kwargs: Any,
     ) -> None:
         """Initialize RedisVectorStore.
 
-        For index arguments that can be passed to RediSearch, see
-        https://redis.io/docs/stack/search/reference/vectors/
-
-        The index arguments will depend on the index type chosen. There
-        are two available index types
-            - FLAT: a flat index that uses brute force search
-            - HNSW: a hierarchical navigable small world graph index
-
         Args:
-            index_name (str): Name of the index.
-            index_prefix (str): Prefix for the index. Defaults to "llama_index".
-                The actual prefix used by Redis will be
-                "{index_prefix}{prefix_ending}".
-            prefix_ending (str): Prefix ending for the index. Be careful when
-                changing this: https://github.com/jerryjliu/llama_index/pull/6665.
-                Defaults to "/vector".
-            index_args (Dict[str, Any]): Arguments for the index. Defaults to None.
-            metadata_fields (List[str]): List of metadata fields to store in the index
-                (only supports TAG fields).
-            redis_url (str): URL for the redis instance.
+            schema (IndexSchema, optional): Redis index schema object.
+            redis_client (Redis, optional): Redis client connection.
+            redis_url (str, optional): Redis server URL.
                 Defaults to "redis://localhost:6379".
-            overwrite (bool): Whether to overwrite the index if it already exists.
+            overwrite (bool, optional): Whether to overwrite the index if it already exists.
                 Defaults to False.
-            kwargs (Any): Additional arguments to pass to the redis client.
 
         Raises:
             ValueError: If redis-py is not installed
             ValueError: If RediSearch is not installed
 
-        Examples:
-            >>> from llama_index.core.vector_stores.redis import RedisVectorStore
-            >>> # Create a RedisVectorStore
-            >>> vector_store = RedisVectorStore(
-            >>>     index_name="my_index",
-            >>>     index_prefix="llama_index",
-            >>>     index_args={"algorithm": "HNSW", "m": 16, "ef_construction": 200,
-                "distance_metric": "cosine"},
-            >>>     redis_url="redis://localhost:6379/",
-            >>>     overwrite=True)
+        # TODO -- update example
```

Comment:
For this to show up properly in our API docs, can we move this entire docstring to under the class definition?

Here's a nice example of what this will end up looking like in our api reference 

https://docs.llamaindex.ai/en/latest/api_reference/storage/vector_store/qdrant/

Reply:
Noted!